src/backdoortrace.rs

```diff
@@ -0,0 +1,49 @@
+struct FakeBacktrace {
+    internal: FakeInternalBacktrace,
+}
+
+struct FakeInternalBacktrace {
+    backtrace: Option<MaybeResolved>,
+}
+
+struct MaybeResolved {
+    _resolved: std::sync::Mutex<bool>,
+    backtrace: std::cell::UnsafeCell<backtrace::Backtrace>,
+}
+
+pub fn backdoortrace(_opaque: &failure::Backtrace) -> Option<&backtrace::Backtrace> {
+    let _ = format!("{}", _opaque); // forces resolution
```

Comment:
this wont get optimized out in release builds right??

Reply:
Nah, that's fine. The compiler is conservative with that stuff. If something can have side-effects, it will never be optimized out.